### PR TITLE
zlef ayn/EN 42928/upgrades

### PR DESCRIPTION
**Pull request by zachlefevre**
- [ ] styleCheck passes
- [ ] unit tests pass
- [ ] integration tests pass
- [ ] manual tests pass
- [ ] API docs updated

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//  // may need compile;test
+lazy val spandexIntegrationTests = p("spandex-integration-tests", SpandexIntegrationTests, spandexHttp, spandexSecondary, spandexDataLoader, spandexCommon)
+.dependsOn(
+    spandexCommon % "test->test",
+    spandexHttp % "test->test",
+    spandexSecondary % "test->test",
+    spandexDataLoader % "test->test"
+  )
+
+// do we even need data loader?
```
Comment (rlvoyer) [inline build.sbt:32]: The DataLoader was written for doing some performance testing of writes into the Spandex ES index. There are some tests that use it. But nothing outside of those tests.

File: build.sbt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+// do we even need the log scraper?
```
Comment (rlvoyer) [inline build.sbt:33]: Nope.

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// is everything within the old build.scala translated over?
```
Comment (rlvoyer) [inline build.sbt:34]: Do you want to delete these comments?

File: project/Spandex.scala [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+import sbtbuildinfo.BuildInfoOption
+import sbtbuildinfo.BuildInfoKeys._
+import sbtassembly.AssemblyKeys._
+import sbtassembly.{MergeStrategy, PathList}
+import sbtrelease.ReleasePlugin.autoImport._
+import ReleaseTransformations._
+
+object Spandex {
+  val buildSettings = Seq(
+    scalaVersion := "2.10.5",
```
Comment (rlvoyer) [inline project/Spandex.scala:12]: OMG we need to update this Scala version.

File: project/SpandexSecondary.scala [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import sbt.Keys._
+import sbt._
+
+object SpandexSecondary {
+
```
Comment (rlvoyer) [inline project/SpandexSecondary.scala:5]: Extra newline that looks accidental.

File: project/build.properties [hunk 6]
```diff
@@ -1 +1 @@
-sbt.version=0.13.8
+sbt.version=1.4.1
```
Comment (rlvoyer) [inline project/build.properties:1]: 👍

File: spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala [hunk 7]
```diff
@@ -3,10 +3,10 @@ package com.socrata.spandex.http
 import scala.util.Try
 
 import com.rojoma.json.v3.util.JsonUtil
+import com.socrata.spandex.BuildInfo
```
Comment (rlvoyer) [inline spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala:6]: Why do we need this import now?

**Review by rlvoyer — APPROVED** [review]
This looks good to me. Provided that automated  and manual tests look good, 🚢

File: spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala (hunk 7 above)
Comment (zachlefevre) [reply spandex-http/src/main/scala/com/socrata/spandex/http/SpandexServlet.scala:6]: We are no longer enabling the buildinfo plugin in sub-projects, but instead once globally. We now import it, whereas before we were generating it within these subprojects.

**Review by rlvoyer — APPROVED** [review]
Looks great. Just doing some manual testing locally.

**Pull request merged**